src/mlpack/bindings/R/print_type_doc_impl.hpp

```diff
@@ -0,0 +1,164 @@
+  {
+    return "A floating-point number (i.e., \"0.5\").";
+  }
+  // A string.
+  else if (std::is_same<T, std::string>::value)
+  {
+    return "A character string (i.e., \"hello\").";
+  }
+  // Not sure what it is...
+  else
+  {
+    throw std::invalid_argument("unknown parameter type " + data.cppType);
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a vector.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& data,
+    const typename std::enable_if<util::IsStdVector<T>::value>::type*)
+{
+  if (std::is_same<T, std::vector<int>>::value)
+  {
+    return "A vector of integers; i.e., c(0, 1, 2).";
+  }
+  else if (std::is_same<T, std::vector<std::string>>::value)
+  {
+    return "A vector of strings; i.e., c(\"hello\", \"goodbye\").";
+  }
+  else
+  {
+    throw std::invalid_argument("unknown vector type " + data.cppType);
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a matrix option.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& data,
+    const typename std::enable_if<arma::is_arma_type<T>::value>::type*)
+{
+  if (std::is_same<typename T::elem_type, double>::value)
+  {
+    if (T::is_col || T::is_row)
+    {
+      return "A 1-d matrix-like containing 'numeric' data (could be an "
+          "'matrix' or 'data.frame' with one dimension of size 1).";
+    }
+    else
+    {
+      return "A 2-d matrix-like containing 'numeric' data (could be an "
+          "'matrix' or a 'data.frame' or anything convertible to an "
+          "2-d 'matrix'.";
+    }
+  }
+  else if (std::is_same<typename T::elem_type, size_t>::value)
+  {
+    if (T::is_col || T::is_row)
+    {
+      return "A 1-d matrix-like containing 'integer' data (could be an "
+          "'matrix' or 'data.frame' with one dimension of size 1).";
+    }
+    else
+    {
+      return "A 2-d matrix-like containing 'integer' data (could be an "
+          "'matrix' or a 'data.frame' or anything convertible to an "
+          "2-d 'matrix'.";
+    }
+  }
+  else
+  {
+    throw std::invalid_argument("unknown matrix type " + data.cppType);
+  }
+}
+
+/**
+ * Return a string representing the command-line type of a matrix tuple option.
+ */
+template<typename T>
+std::string PrintTypeDoc(
+    util::ParamData& /* data */,
+    const typename std::enable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>::value>::type*)
+{
+  return "A 2-d array containing 'numeric' data.  Like the regular 2-d matrices"
+      ", this can be a 'matrix', or a data.frame. However, this type can also "
+      "accept a data.frame that has columns of type 'character', 'logical' or "
+      "'factor'.  These values will be converted to numeric indices before "
+      "being passed to mlpack, and then inside mlpack they will be properly "
+      "treated as categorical variables, so there is no need to do one-hot "
+      "encoding for this matrix type.";
```

Comment:
Nice explanation!  Always great when we can advertise that one-hot encoding isn't necessary, since it's a pain (both implementationally and computationally :smile:)